src/main/java/uk/ac/cam/cl/dtg/isaac/quiz/IQuestionAttemptManager.java

```diff
@@ -81,13 +81,15 @@ void mergeAnonymousQuestionInformationWithRegisteredUserRecord(String anonymousU
       throws SegueDatabaseException;
 
   /**
-   * Count the users by role which have answered questions over the previous time interval.
+   * Retrieves a count of users by their roles who have answered questions within the specified time ranges.
    *
-   * @param timeInterval time interval over which to count
-   * @return map of counts for each role
-   * @throws SegueDatabaseException - if there is a problem with the database.
+   * @param timeIntervals An array of time ranges (in string format) for which to get the user counts.
+   *                      Each time range is used in the SQL query to filter the results.
+   * @return A map where the keys are the time ranges and the values are another map containing the count of users
+   *     for each role within that time range.
+   * @throws SegueDatabaseException If there is a database-related issue, such as a SQL exception.
    */
-  Map<Role, Long> getAnsweredQuestionRolesOverPrevious(TimeInterval timeInterval)
+  Map<TimeInterval, Map<Role, Long>> getAnsweredQuestionRolesOverPrevious(TimeInterval[] timeIntervals)
```

Comment:
these time interval strings...how are they constructed and what's their origin? an array of strings seems like a weird thing when you could use an array of date objects/epochs/something else. Not coming from any end-user input or anything?

Reply:
We're using an array of TimeInterval enums to make it really easy to manage the specific time ranges we're interested in for our database queries. For example, in the function addRangedAnsweredQuestionStats, we've defined an array with time intervals like SEVEN_DAYS, THIRTY_DAYS, and NINETY_DAYS.
This approach makes our code cleaner, easier to understand, and safer because we're sticking to predefined, valid options.

Reply:
I have changed the comment to reflect that its looking at an array of enums